components/infobox/commons/custom/infobox_person_user_custom.lua

```diff
@@ -6,98 +6,67 @@
 -- Please see https://github.com/Liquipedia/Lua-Modules to contribute
 --
 
+local Array = require('Module:Array')
 local Class = require('Module:Class')
 local Lua = require('Module:Lua')
 local String = require('Module:StringUtils')
 
 local Injector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
-local User = Lua.import('Module:Infobox/Person', {requireDevIfEnabled = true})
+local User = Lua.import('Module:Infobox/Person/User', {requireDevIfEnabled = true})
 
 local Widgets = require('Module:Infobox/Widget/All')
 local Cell = Widgets.Cell
 local Title = Widgets.Title
 local Center = Widgets.Center
 
-local CustomUser = Class.new()
+---@class CustomInfoboxUser: InfoboxUser
+local CustomUser = Class.new(User)
 
 local CustomInjector = Class.new(Injector)
 
-local _args
-
+---@param frame Frame
+---@return Html
 function CustomUser.run(frame)
-	local user = User(frame)
-	user.args.informationType = user.args.informationType or 'User'
-	_args = user.args
-
-	user.shouldStoreData = CustomUser.shouldStoreData
-	user.getStatusToStore = CustomUser.getStatusToStore
-	user.getPersonType = CustomUser.getPersonType
+	local user = CustomUser(frame)
+	user:setWidgetInjector(CustomInjector(user))
 
-	user.createWidgetInjector = CustomUser.createWidgetInjector
+	user.args.informationType = user.args.informationType or 'User'
 
 	return user:createInfobox()
 end
 
+---@param id string
+---@param widgets Widget[]
+---@return Widget[]
 function CustomInjector:parse(id, widgets)
```

Comment:
All of this can be moved to the infobox/person/user right?

Reply:
needs to be completely overwritten in other customs anyways since customInjector is different there (new instance)

Reply:
Hm true, don't like it but OK